test/unit/cute/intel_xe/copy_block.cpp

```diff
@@ -0,0 +1,1817 @@
+                                                              16, 16),
+        Layout<Shape<_1, _16>, Stride<_0, _1>>{},
+        Layout<Shape<_8, _1>, Stride<_1, _0>>{});
+    static constexpr auto subgroup_size = 16;
+    auto blockDim = syclcompat::dim3(size(tiled_load));
+//
+// Launch the kernel
+//
+    syclcompat::experimental::launch<
+        copy_kernel_vectorized<decltype(S), decltype(D), decltype(tiled_load),
+                               decltype(tiled_store), XE_2D_U16x8x32_LD_N>,
+        subgroup_size>(1, blockDim, S, D, tiled_load, tiled_store);
+
+    syclcompat::wait_and_throw();
+    syclcompat::memcpy<dtype>(host_output.data(), device_output, M * N);
+
+    for (int i = 0; i < 16; ++i) {
+      for (int j = 0; j < 16; ++j) {
+        // printf("%d  %d\n", int(h_in[i]), int(h_out[i]));
+        EXPECT_EQ(host_output[i * 16 + j],
+                  host_src[(i % 8) * 32 + j + (i / 8) * 16]);
+      }
+    }
+    print("XE_2D_U16x8x32_LD_N test end \n\n");
+  }
+
+  {
+    print("XE_2D_U16x16x32_LD_N test: \n");
+    constexpr int M = 16;
+    constexpr int N = 32;
+    using dtype = uint16_t;
+    //
+    // Allocate and initialize
+    //
+    std::vector<dtype> host_src(M * N);
+    std::vector<dtype> host_output(M * N);
+
+    dtype *device_src = syclcompat::malloc<dtype>(M * N);
+    dtype *device_output = syclcompat::malloc<dtype>(M * N);
+
+    for (size_t i = 0; i < host_src.size(); ++i) {
+      host_src[i] = static_cast<dtype>(i);
+    }
+
+    syclcompat::memcpy<dtype>(device_src, host_src.data(), M * N);
+    syclcompat::memcpy<dtype>(device_output, host_output.data(), M * N);
+
+    Tensor S =
+        make_tensor(make_gmem_ptr(device_src),
+                    make_layout(Shape<Int<M>, Int<N>>{}, Stride<Int<N>, _1>{}));
+    Tensor D = make_tensor(
+        make_gmem_ptr(device_output),
+        make_layout(Shape<Int<32>, Int<16>>{}, Stride<Int<16>, _1>{}));
+
+    auto tiled_load = make_tiled_copy(
+        Copy_Atom<Copy_Traits<XE_2D_U16x16x32_LD_N>, uint16_t>{}.with(device_src, N, M,
+                                                               N),
+        Layout<Shape<_1, _16>, Stride<_0, _1>>{},
+        Layout<Shape<_16, _2>, Stride<_1, _16>>{});
+    auto tiled_store = make_tiled_copy(
+        Copy_Atom<Copy_Traits<XE_2D_U16x8x16_ST_N>, uint16_t>{}.with(device_output, 16,
+                                                              32, 16),
+        Layout<Shape<_1, _16>, Stride<_0, _1>>{},
+        Layout<Shape<_8, _1>, Stride<_1, _0>>{});
+    static constexpr auto subgroup_size = 16;
+    auto blockDim = syclcompat::dim3(size(tiled_load));
+//
+// Launch the kernel
+//
+    syclcompat::experimental::launch<
+        copy_kernel_vectorized<decltype(S), decltype(D), decltype(tiled_load),
+                               decltype(tiled_store), XE_2D_U16x16x32_LD_N>,
+        subgroup_size>(1, blockDim, S, D, tiled_load, tiled_store);
```

Comment:
Wrong usage of `syclcompat::launch` function, I suspect this would not compile with the latest nightly